analysis/model/fn-prepare_model_input.R

```diff
@@ -0,0 +1,112 @@
+prepare_model_input <- function(name) {
+  # Load active analyses ---------------------------------------------------------
+  print("Load active analyses")
+
+  active_analyses <- readr::read_rds("lib/active_analyses.rds")
+
+  # Filter active_analyses to model inputs to be prepared ------------------------
+  print("Filter active_analyses to model inputs to be prepared")
+
+  active_analyses <- active_analyses[active_analyses$name == name, ]
+
+  if (nrow(active_analyses) == 0) {
+    stop(paste0("Input: ", name, " does not match any analyses"))
+  }
+
+  # Load data ------------------------------------------------------------------
+  print(paste0("Load data for ", active_analyses$name))
+
+  input <- dplyr::as_tibble(readr::read_rds(paste0(
+    "output/dataset_clean/input_",
+    active_analyses$cohort,
+    "_clean.rds"
+  )))
+
+  # Restrict to required variables for dataset preparation ---------------------
+  print("Restrict to required variables for dataset preparation")
+
+  active_analyses$covariate_sex <- dplyr::if_else(
```

Comment:
We don't want to overwrite `active_analyses$covariate_sex` and `active_analyses$covariate_other` as then the variables get passed through to `keep`, and we don't want them in the final dataset.

Instead, we should make a new vector for additional variables that you call in the list on line 39. The new vector should be NULL unless the name parameter indicates that the analysis requires an additional variable.